components/admin_console/custom_plugin_settings/index.ts

```diff
@@ -17,24 +19,26 @@ import {it} from '../admin_definition';
 
 import {appsPluginID} from 'utils/apps';
 
-import CustomPluginSettings from './custom_plugin_settings.jsx';
+import CustomPluginSettings from './custom_plugin_settings';
 import getEnablePluginSetting from './enable_plugin_setting';
 
+type OwnProps = { match: { params: { plugin_id: string } } }
+
 function makeGetPluginSchema() {
     return createSelector(
         'makeGetPluginSchema',
-        (state, pluginId) => state.entities.admin.plugins[pluginId],
-        (state, pluginId) => getAdminConsoleCustomComponents(state, pluginId),
+        (state: GlobalState, pluginId: string) => state.entities.admin.plugins![pluginId],
+        (state: GlobalState, pluginId: string) => getAdminConsoleCustomComponents(state, pluginId),
         (state) => appsFeatureFlagEnabled(state),
-        (plugin, customComponents, appsFeatureFlagIsEnabled) => {
+        (plugin: PluginRedux & {translate?: boolean}, customComponents, appsFeatureFlagIsEnabled) => {
```

Comment:
Should `translate` be added to `PluginRedux`? I'm trying to figure out if it's part of that or part of the underlying `PluginManifest` type

Reply:
Unfortunately I couldnt track it down. Please let me know and I can 
- add it to `PluginRedux` type or
- add it to `PluginManifest` type or
- leave as it is

Reply:
Thanks for the comments @michelengelen 

- For `plugin`, I added `{translate?: boolean}` because on line 68, `plugin.translate` is used. Therefore the question is 
should I keep `{translate?: boolean}` here or or add it to `PluginRedux` or `PluginManifest`

- For `customComponents`, the type is actually ` Record<string, AdminConsolePluginComponent>` so I updated according. Thanks for pointing me to correct place

Reply:
>     * For `plugin`, I added `{translate?: boolean}` because on line 68, `plugin.translate` is used. Therefore the question is
>       should I keep `{translate?: boolean}` here or or add it to `PluginRedux` or `PluginManifest`

I would be OK with adding `{translate?: boolean}` to the `Plugin` (which is basically the same as the `PluginManifest`) type. It is in the same file where the `PluginRedux` type is defined.
 
>     * For `customComponents`, the type is actually ` Record<string, AdminConsolePluginComponent>` so I updated according. Thanks for pointing me to correct place

That's strange since the selector returns only the custom component for a specific plugin as seen here:
https://github.com/mattermost/mattermost-webapp/blob/ae69876e2c30c46a932807e0fb44b6d98ef6250a/selectors/admin_console.jsx#L26-L27

Reply:
`{translate?: boolean}` is added to `Plugin`

For `customComponents`, I see the following when I `console.log` `customComponents` in `index`. What am I missing here?
<img width="495" alt="Screen Shot 2022-04-14 at 23 21 12" src="https://user-images.githubusercontent.com/37421564/163478707-26b314a7-a170-437d-b958-cc48d00bdf17.png">

Reply:
I am not sure I follow what you are asking here. For me it does seem as if it returns a single `customComponent` so maybe we should first find if there is a case where multiple components get returned and if not rename the function to `getAdminConsoleCustomComponent` to make it more visible that it is expected to only return a single component

Reply:
Honestly, I wouldn't be surprised if `state.plugins.adminConsoleCustomComponents` is typed incorrectly. The plugin code was written in a very not-TS-friendly way, and I had a hell of a time figuring out the current type definitions when I added those after the fact. Let me see if I can find what its type actually should be

Reply:
Thanks to both of you for guiding me on this